osconfig_tests/gcp_clients/gcp_clients.go

```diff
@@ -28,37 +29,57 @@ import (
 
 var (
 	storageClient  *storage.Client
+	computeClient  compute.Client
 	osconfigClient *osconfig.Client
 
 	populateClientOnce sync.Once
 )
 
-// PopulateClients creates singleton clients for various services
-func PopulateClients(ctx context.Context) {
+func populateClients(ctx context.Context) error {
+	var err error
 	populateClientOnce.Do(func() {
-		createOsConfigClient(ctx)
-		createStorageClient(ctx)
+		err = createComputeClient(ctx)
+		err = createOsConfigClient(ctx)
+		err = createStorageClient(ctx)
```

Comment:
since these are executed sequentially, won't each call overwrite `err` from the previous invocation?

Reply:
I really didn't like this setup, made this much simpler